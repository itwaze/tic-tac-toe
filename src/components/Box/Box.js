import React from "react";
import styled from "styled-components";

const StyledDiv = styled.div`
	display: flex;
	flex-wrap: wrap;
	align-items: center;
	justify-content: ${({ justify }) => justify};
	margin: ${({ margin }) => margin};
	width: ${({ width }) => width};
`;

const Box = (props) => {
	const { children, ...rest } = props;
	return <StyledDiv {...rest}>{children}</StyledDiv>;
};

export default Box;
