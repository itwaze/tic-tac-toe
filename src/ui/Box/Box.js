import React from "react";
import styled, { css } from "styled-components";

const StyledDiv = styled.div`
  display: flex;
  flex-wrap: wrap;
  flex-direction: ${({ direction }) => direction};
  align-items: center;
  justify-content: ${({ justify }) => justify};
  margin: ${({ margin }) => margin};
  width: ${({ width }) => width};
  height: ${({ height }) => height};
  padding: ${({ padding }) => padding};
  box-sizing: border-box;
  ${({ border }) =>
    border &&
    css`
      border: 1px solid #ccc;
    `};
`;

const Box = (props) => {
  const { children, ...rest } = props;
  return <StyledDiv {...rest}>{children}</StyledDiv>;
};

export default Box;
