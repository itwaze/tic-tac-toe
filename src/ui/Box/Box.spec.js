import React from "react";
import { render } from "@testing-library/react";

import Box from "./Box";

describe("Box", () => {
  it("should take a snapshot", () => {
    const { asFragment } = render(<Box border />);
    expect(asFragment()).toMatchSnapshot();
  });
});
