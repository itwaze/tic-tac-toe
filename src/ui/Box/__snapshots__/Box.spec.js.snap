// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ djiRpn"
  />
</DocumentFragment>
`;
